src/pixelator/analysis/__init__.py

```diff
@@ -137,8 +137,14 @@ def analyse_pixels(
         )
     }
     # save dataset
-    dataset.save(str(Path(output) / f"{output_prefix}.analysis.dataset.pxl"))
+    dataset.save(str(Path(output) / f"{output_prefix}.dataset.pxl"))
```

Comment:
I will leave it up to Johan to make the interface stricter in all the PixelDataset classes. I have just removed the str conversion calls here since that is not needed